### PR TITLE
Update CRD comment and generate crd.md

### PR DESCRIPTION
**Pull request by moonsub-kim**
[crd.md](https://github.com/moonsub-kim/flink-on-k8s-operator/blob/update-comment/docs/crd.md)

### CODE REVIEW

File: api/v1beta1/flinkcluster_types.go [hunk 1]
```diff
@@ -487,40 +525,42 @@ type BatchSchedulerSpec struct {
...
 
-	// Flink JobManager spec.
+	// _(Required)_ Flink JobManager spec.
 	JobManager JobManagerSpec `json:"jobManager"`
 
-	// Flink TaskManager spec.
+	// _(Required)_ Flink TaskManager spec.
 	TaskManager TaskManagerSpec `json:"taskManager"`
 
-	// (Optional) Job spec. If specified, this cluster is an ephemeral Job
```
Comment (regadas) [inline api/v1beta1/flinkcluster_types.go:513]: we can keep `Optional` now ... it's easy to maintain now

**Review by regadas — COMMENTED** [review]
Thanks for doing this @moonsub-kim! 

Just left a small comment around the `Required` vs `Optional` but otherwise this looks great!

File: api/v1beta1/flinkcluster_types.go (hunk 1 above)
Comment (moonsub-kim) [reply api/v1beta1/flinkcluster_types.go:513]: Can I add `optional` mark into all the omitted fields except to fill the default value?
These include not only job, taskmanager.port but volume, volumemount, initcontainers, and so on..

Or is there something you are thinking about like excluding the k8s resource fields?

Comment (regadas) [reply api/v1beta1/flinkcluster_types.go:513]: Yeah we can add the `optional` mark to everything that is really optional.

Comment (moonsub-kim) [reply api/v1beta1/flinkcluster_types.go:513]: I have updated it~!

**Review by regadas — APPROVED** [review]
Thanks for doing this @moonsub-kim

**Pull request merged**